keps/2937-resource-transformer/README.md

```diff
@@ -0,0 +1,560 @@
+    Message:               couldn't assign flavors to pod set main: insufficient unused quota for kueue.x-k8s.io/accelerator-memory in flavor default-flavor, 20G more needed
+    Observed Generation:   1
+    Reason:                Pending
+    Status:                False
+    Type:                  QuotaReserved
+```
+
+We could improve the observability by extending the Workload Status with a ResourceRequests
+field that would contain a copy of the `TotalRequests` field of the controller-internal `workload.Info`
+struct.  If we did this, we would instead see something like the below for a non-admitted Workload:
+```yaml
+Spec:
+  Active:  true
+  Pod Sets:
+    Count:  1
+    Name:   main
+    Template:
+      Metadata:
+      Spec:
+        Containers:
+          ...
+          Resources:
+            Limits:
+              nvidia.com/mig-1g.5gb:   2
+              nvidia.com/mig-2g.10gb:  1
+            Requests:
+              Cpu:                     1
+              Memory:                  100G
+              nvidia.com/mig-1g.5gb:   2
+              nvidia.com/mig-2g.10gb:  1
+...
+Status:
+  Conditions:
+    Last Transition Time:  2024-09-19T14:59:01Z
+    Message:               couldn't assign flavors to pod set main: insufficient unused quota for kueue.x-k8s.io/accelerator-memory in flavor default-flavor, 20G more needed
+    Observed Generation:   1
+    Reason:                Pending
+    Status:                False
+    Type:                  QuotaReserved
+  ResourceRequests:
+    Cpu:                                1
+    kueue.x-k8s.io/accelerator-memory:  20G
+    Memory:                             97656250Ki
+```
+
+### Test Plan
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+None
+
+#### Unit Tests
+
+Unit tests of the helper function invoked from `workload.NewInfo` will be added.
+
+Unit tests of loading the `ResourceTransformationMatrix` from a `Configuration` will be added.
+
+#### Integration tests
+
+Because resource transformations are configured during manager startup, we need to run
+an entire integration test suite with a non-empty resource transformation if we want to
+do any integration testing with resource transformations. I'd propose to extend
+`test/integration/controller/core` by adding a resource transformation to the manager that
+maps from a synthetic extended resource to both `cpu` and `credits`.  We'd then add a
+`test/integration/controller/core/resource_transform_test.go` that would define a cluster
+queue with a resource group that covers `cpu` and `credit` and run a few quota and queuing
+tests against it.
+
+### Graduation Criteria
+
+## Implementation History
+
+## Drawbacks
+
+Adds an additional dimension of configuration that needs to be documented
+and maintained.
+
+## Alternatives
```

Comment:
Move whatever is discarded here, with subtitles to distinguish the different decision points.